docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,605 @@ under the License.
 
 # Glossary
 
+#### Aggregation
+
+Aggregation is an operation that takes multiple values and returns a single value. When working with 
+streams, it generally makes more sense to think in terms of aggregations over finite windows, rather 
+than over the entire stream.
+
+#### (Flink) Application
+
+A Flink application is any user program that submits one or multiple [Flink Jobs](#flink-job) from its
+`main()` method. The execution of these jobs can happen in a local JVM or on a remote setup of clusters 
+with multiple machines.
+
+The jobs of an application can either be submitted to a long-running [Session Cluster](#session-cluster),
+to a dedicated [Application Cluster](#application-cluster), or to a [Job Cluster](#job-cluster).
+
+#### Application Cluster
+
+A Flink application cluster is a dedicated [Flink cluster](#(flink)-cluster) that only executes 
+[Flink jobs](#flink-job) from one [Flink application](#(flink)-application). The lifetime of the Flink
+cluster is bound to the lifetime of the Flink application.
+
+#### Asynchronous Snapshotting
+
+A form of [snapshotting](#snapshot) that doesn't impede the ongoing stream processing by allowing an 
+operator to continue processing while it stores its state snapshot, effectively letting the state 
+snapshots happen asynchronously in the background.
+
+#### At-least-once
+
+A fault-tolerance guarantee and data delivery approach where multiple attempts are made at delivering
+an event such that at least one succeeds. This guarantees that nothing is lost, but you may experience 
+duplicated results.
+
+#### At-most-once
+
+A data delivery approach where each event is delivered zero or one times. There is lower latency but
+events may be lost.
+
+#### Backpressure
+
+A situation where a system is receiving data at a higher rate than it can process during a temporary 
+load spike.
```

Comment:
it's not necessarily a temporary spike; it could just be that the cluster doesn't have sufficient resources